### PR TITLE
Bug 1917336: add UEFI instructions when using PXE

### PR DESCRIPTION
**Pull request by miabbott**

### CODE REVIEW

**Comment by miabbott** [comment]
I basically copied what was in the 4.5 docs for configuring/booting UEFI over PXE and adapted it to the new "live" format artifacts we produce.

I haven't actually verified the instructions work as written, so YMMV.

cc: @bgilbert @mburke5678

File: modules/installation-user-infra-machines-pxe.adoc [hunk 1]
```diff
@@ -103,6 +103,40 @@ or other boot options.
...
+# umount /mnt/{efiboot,iso}
+----
+
+.. Copy the `EFI/redhat/grub.cfg` file that is included in the {op-system} ISO to your TFTP server.
+
+.. Edit the `grub.cfg` file to include the following arguments:
++
+----
+menuentry 'Install Red Hat Enterprise Linux CoreOS' --class fedora --class gnu-linux --class gnu --class os {
+	linuxefi rhcos-<version>-live-kernel-<architecture> nomodeset rd.neednet=1 coreos.inst=yes coreos.inst.install_dev=/dev/sda coreos.inst.rootfs_url=http://<HTTP_server>/rhcos-<version>-live-rootfs.<architecture>.img coreos.inst.ignition_url=http://<HTTP_server>/bootstrap.ign <1>
```
Comment (bgilbert) [inline modules/installation-user-infra-machines-pxe.adoc:128]: There are some stale arguments here:
```suggestion
	linuxefi rhcos-<version>-live-kernel-<architecture> coreos.inst.install_dev=/dev/sda coreos.inst.rootfs_url=http://<HTTP_server>/rhcos-<version>-live-rootfs.<architecture>.img coreos.inst.ignition_url=http://<HTTP_server>/bootstrap.ign <1>
```

**Comment by mburke5678** [comment]
@miabbott Any updates on this?

**Comment by miabbott** [comment]
> @miabbott Any updates on this?

Was on PTO last week; still haven't had a chance to test this configuration.

**Comment by miabbott** [comment]
Using UEFI to boot via PXE is hairy, but this works.

**Comment by mburke5678** [comment]
@miabbott Are these changes for 4.6+? We might (meaning me) need to pull separate PRs for each branch due to changes to the installation files. At least one for 4.6 and 4.7 and another for 4.8 and master. Significant changes were made for 4.8+.

**Comment by miabbott** [comment]
@mburke5678 Yup, these changes apply to each version all the way back to 4.6

**Comment by mburke5678** [comment]
@miabbott OK. I will move your changes into separate PRs. Thanks for your help with this!!

**Comment by mburke5678** [comment]
@miabbott PTAL. I added the step from your PR and made it match 4.5.

**Comment by miabbott** [comment]
@mburke5678 I made one small tweak to the final commit that should be changed in the other PRs, but otherwise they look fine.

**Comment by mburke5678** [comment]
@miabbott Thanks! 

You [added this step to 4.5 only](https://github.com/openshift/openshift-docs/pull/27120/files#diff-3e2e52939307b0dbc8d700d3f8ddb0b4ef35b3271f5a8b4fee21ada94b03a176R99-R101). Should this step be in the other versions also?

**Comment by miabbott** [comment]
> @miabbott Thanks!
> 
> You [added this step to 4.5 only](https://github.com/openshift/openshift-docs/pull/27120/files#diff-3e2e52939307b0dbc8d700d3f8ddb0b4ef35b3271f5a8b4fee21ada94b03a176R99-R101). Should this step be in the other versions also?

That's implied here - https://github.com/openshift/openshift-docs/pull/33051/files#diff-3e2e52939307b0dbc8d700d3f8ddb0b4ef35b3271f5a8b4fee21ada94b03a176R191-R197

But you are right that those steps are required for any PXE installation.  Let me massage this PR one more time.

**Comment by miabbott** [comment]
Another update :arrow_up:

**Comment by miabbott** [comment]
ugh, found one more nit I had to fix :arrow_up:

**Comment by mburke5678** [comment]
@miabbott Should i remove the _For more information on how to configure a PXE server for UEFI boot,_ for iPXE?

**Comment by miabbott** [comment]
> @miabbott Should i remove the _For more information on how to configure a PXE server for UEFI boot,_ for iPXE?

I don't think so.  Setting up a PXE server will allow you to serve up both artifacts like the EFI binaries, kernel, initramfs and the iPXE scripts that can be used.

Since this kind of PXE configuration can be challenging (I know from experience!), I think giving additional resources for users would be useful.

**Pull request closed without merging**